ethdb/kv_remote2.go

```diff
@@ -0,0 +1,400 @@
+package ethdb
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/ledgerwatch/turbo-geth/common"
+	"github.com/ledgerwatch/turbo-geth/ethdb/remote"
+	"github.com/ledgerwatch/turbo-geth/log"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/test/bufconn"
+)
+
+// generate the messages
+//go:generate protoc --go_out=. "./remote/kv.proto"
+
+// generate the services
+//go:generate protoc --go-grpc_out=. "./remote/kv.proto"
+
+type remote2Opts struct {
+	DialAddress string
+	inMemConn   *bufconn.Listener
+}
+
+type Remote2KV struct {
+	opts   remote2Opts
+	client remote.KvClient
+	conn   *grpc.ClientConn
+	log    log.Logger
+}
+
+type remote2Tx struct {
+	ctx context.Context
+	db  *Remote2KV
+
+	handle     uint64
+	finalizers []context.CancelFunc
+}
+
+type remote2Bucket struct {
+	tx           *remote2Tx
+	bucketHandle uint64
+	initialized  bool
+	name         []byte
+}
+
+type remote2Cursor struct {
+	ctx          context.Context
+	bucket       *remote2Bucket
+	cursorHandle uint64
+	initialized  bool
+	prefetch     uint32
+	prefix       []byte
+
+	stream             remote.Kv_SeekClient
+	streamClose        context.CancelFunc
+	streamingRequested bool
+}
+
+type remote2NoValuesCursor struct {
+	*remote2Cursor
+}
+
+func (opts remote2Opts) ReadOnly() remote2Opts {
+	return opts
+}
+
+func (opts remote2Opts) Path(path string) remote2Opts {
+	opts.DialAddress = path
```

Comment:
I'm not sure, but it looks like it shouldn't work because of the received by value. The same for below.

Reply:
it returns because on next line it returns opts